File: rps-ts-server/src/application/usecase/RunGame.ts
import { determineWinner } from "../../domain/Game";
import { Move } from "../../domain/Move";
import { Result } from "../../domain/Result";
import { IGame } from "../../domain/repository/Game";

export class Game implements IGame {
    play(player1Move: Move, player2Move: Move): Result {
        return determineWinner(player1Move, player2Move);
    }
}

File: rps-go-server/internal/app/application/usecase/RunGame.go
package usecase

import (
	"rock-paper-scissors/internal/app/domain"
	"rock-paper-scissors/internal/app/domain/repository"
)

type RunGameArgs struct {
	Player1Move    domain.Move
	Player2Move    domain.Move
	GameRepository repository.IGame
}

func RunGame(args RunGameArgs) string {
	return args.GameRepository.DetermineWinner(args.Player1Move, args.Player2Move)
}
